apps/server/src/modules/tool/tool-launch/service/auto-parameter-strategy/auto-context-name.strategy.ts

```diff
@@ -0,0 +1,55 @@
+import { ColumnBoardService } from '@modules/board';
+import { CourseService } from '@modules/learnroom';
+import { Injectable } from '@nestjs/common';
+import { BoardExternalReferenceType, ColumnBoard, Course, EntityId } from '@shared/domain';
+import { CustomParameterType, ToolContextType } from '../../../common/enum';
+import { ContextExternalTool } from '../../../context-external-tool/domain';
+import { SchoolExternalTool } from '../../../school-external-tool/domain';
+import { ParameterTypeNotImplementedLoggableException } from '../../error';
+import { AutoParameterStrategy } from './auto-parameter.strategy';
+
+@Injectable()
+export class AutoContextNameStrategy implements AutoParameterStrategy {
+	constructor(private readonly courseService: CourseService, private readonly columnBoardService: ColumnBoardService) {}
+
+	async getValue(
+		schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalTool
+	): Promise<string | undefined> {
+		switch (contextExternalTool.contextRef.type) {
+			case ToolContextType.COURSE: {
+				const courseValue: string = await this.getCourseValue(contextExternalTool.contextRef.id);
+
+				return courseValue;
+			}
+			case ToolContextType.BOARD_ELEMENT: {
+				const boardValue: string | undefined = await this.getBoardValue(contextExternalTool.contextRef.id);
+
+				return boardValue;
+			}
+			default: {
+				throw new ParameterTypeNotImplementedLoggableException(
+					`${CustomParameterType.AUTO_CONTEXTNAME}/${contextExternalTool.contextRef.type as string}`
+				);
+			}
+		}
+	}
+
+	private async getCourseValue(courseId: EntityId): Promise<string> {
```

Comment:
We should think about introducing an interface e.g. "ExternalToolsNameProvider" which will be implemented by course and columnboard service and returns an independet object.
The interface should also provide a function to return the toolContextType. 
This would decouple our strategy from the services.

The getValue function can use the nestjs reflector to get all implementations of the interface